src/components/CurriculumComponents/UnitsTab/UnitsTab.tsx

```diff
@@ -58,166 +58,178 @@ interface YearSelection {
+        });
+      }
 
-    // Populate list of domain filter values
+      // Populate list of tier filter values
 
-    if (
-      unit.domain &&
-      unit.domain_id &&
-      currentYearData.domains.every((d) => d.domain_id !== unit.domain_id)
-    ) {
-      currentYearData.domains.push({
-        domain: unit.domain,
-        domain_id: unit.domain_id,
-      });
-    }
+      if (
+        unit.tier &&
+        unit.tier_slug &&
+        currentYearData.tiers.every((t) => t.tier_slug !== unit.tier_slug)
+      ) {
+        currentYearData.tiers.push({
+          tier: unit.tier,
+          tier_slug: unit.tier_slug,
+        });
+      }
 
-    // Populate list of tier filter values
+      // Check for duplicate unit slugs
 
-    if (
-      unit.tier &&
-      unit.tier_slug &&
-      currentYearData.tiers.every((t) => t.tier_slug !== unit.tier_slug)
-    ) {
-      currentYearData.tiers.push({
-        tier: unit.tier,
-        tier_slug: unit.tier_slug,
-      });
-    }
-
-    // Check for duplicate unit slugs
+      if (unitSlugs.has(unit.slug)) {
+        duplicateUnitSlugs.add(unit.slug);
+      } else {
+        unitSlugs.add(unit.slug);
+      }
+    });
 
-    if (unitSlugs.has(unit.slug)) {
-      duplicateUnitSlugs.add(unit.slug);
-    } else {
-      unitSlugs.add(unit.slug);
-    }
-  });
+    // Sort year data
 
-  // Sort year data
+    yearOptions.sort((a, b) => Number(a) - Number(b));
 
-  yearOptions.sort((a, b) => Number(a) - Number(b));
+    // Sort threads
 
-  // Sort threads
+    const threadOrders = new Set(threadOptions.map((to) => to.order));
+    if (threadOptions.length > threadOrders.size) {
+      // In secondary science multiple threads can have the same order value due
+      // to multiple subjects (eg biology, chemistry, physics) being shown, so
+      // if orders are not unique, sort alphabetically by slug
 
-  const threadOrders = new Set(threadOptions.map((to) => to.order));
-  if (threadOptions.length > threadOrders.size) {
-    // In secondary science multiple threads can have the same order value due
-    // to multiple subjects (eg biology, chemistry, physics) being shown, so
-    // if orders are not unique, sort alphabetically by slug
+      threadOptions.sort((a, b) => a.slug.localeCompare(b.slug));
+    } else {
+      // If orders are unique, use them to sort
 
-    threadOptions.sort((a, b) => a.slug.localeCompare(b.slug));
-  } else {
-    // If orders are unique, use them to sort
+      threadOptions.sort((a, b) => a.order - b.order);
+    }
 
-    threadOptions.sort((a, b) => a.order - b.order);
-  }
+    // Set up year-specific filters (domains, child subjects, tiers):
+    // populate options and select defaults
 
-  // Set up year-specific filters (domains, child subjects, tiers):
-  // populate options and select defaults
+    const initialYearSelection = {} as YearSelection;
+    Object.keys(yearData).forEach((year) => {
+      const data = yearData[year];
```

Comment:
Maybe call this something else given there is already data in the outer scope and it's in the dependency list of thie useEffect